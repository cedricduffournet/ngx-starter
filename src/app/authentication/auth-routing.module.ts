import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';
import {
  LoginViewComponent,
  SendResetPasswordViewComponent,
  ResetPasswordViewComponent,
  RegisterViewComponent,
  RegisterConfirmViewComponent
} from '@app/authentication/containers';
import { authConfig } from '@app/authentication/auth.config';

const AUTHENTICATION_ROUTES: Routes = [
  { path: 'login', component: LoginViewComponent },
  { path: 'resetting', component: SendResetPasswordViewComponent },
  {
    path: `${authConfig.path.passwordResetConfirm}/:token`,
    component: ResetPasswordViewComponent
  },
  { path: 'register', component: RegisterViewComponent },
  {
    path: `${authConfig.path.registerConfirm}/:token`,
    component: RegisterConfirmViewComponent
  }
];

@NgModule({
  imports: [RouterModule.forChild(AUTHENTICATION_ROUTES)],
  exports: [RouterModule]
})
export class AuthRoutingModule {}
