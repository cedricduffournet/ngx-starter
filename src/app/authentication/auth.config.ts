export const authConfig = {
  path: {
    registerConfirm: 'register/confirm',
    passwordResetConfirm: 'password/reset'
  },
  storageKey: 'authentication'
};
