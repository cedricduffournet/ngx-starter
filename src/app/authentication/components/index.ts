export * from './auth-layout/auth-layout.component';
export * from './login-form/login-form.component';
export * from './register-form/register-form.component';
export * from './reset-password-form/reset-password-form.component';
export * from './send-reset-password-form/send-reset-password-form.component';
