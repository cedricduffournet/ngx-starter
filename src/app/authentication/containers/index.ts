export * from './login-view/login-view.component';
export * from './register-confirm-view/register-confirm-view.component';
export * from './register-view/register-view.component';
export * from './reset-password-view/reset-password-view.component';
export * from './send-reset-password-view/send-reset-password-view.component';
export * from './logout-modal/logout-modal.component';
