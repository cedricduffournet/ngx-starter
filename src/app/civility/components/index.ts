export * from './civility-add/civility-add.component';
export * from './civility-update/civility-update.component';
export * from './civility-delete/civility-delete.component';
export * from './civility-form/civility-form.component';
export * from './civility-item/civility-item.component';
export * from './civility-items/civility-items.component';
