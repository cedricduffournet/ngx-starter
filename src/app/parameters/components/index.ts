export * from './referenced-table-menu/referenced-table-menu.component';
