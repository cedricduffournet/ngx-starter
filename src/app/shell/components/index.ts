export * from './navbar/navbar.component';
export * from './dropdown-navbar/dropdown-navbar.component';
export * from './menu/menu.component';
