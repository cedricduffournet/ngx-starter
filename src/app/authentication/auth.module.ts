import { NgModule } from '@angular/core';

import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';

import { SharedModule } from '@app/shared/shared.module';
import { CivilityModule } from '@app/civility/civility.module';
import { LangSwitcherModule } from '@app/shared/lang-switcher/lang-switcher.module';
import { LogoModule } from '@app/shared/logo/logo.module';
import { AuthRoutingModule } from '@app/authentication/auth-routing.module';
import { AuthEffects } from '@app/authentication/state/effects';
import { reducers } from '@app/authentication/state/reducers';
import { AuthFacade } from '@app/authentication/state/auth.facade';

import {
  LoginViewComponent,
  SendResetPasswordViewComponent,
  ResetPasswordViewComponent,
  RegisterViewComponent,
  RegisterConfirmViewComponent,
  LogoutModalComponent
} from '@app/authentication/containers';

import {
  AuthLayoutComponent,
  LoginFormComponent,
  ResetPasswordFormComponent,
  SendResetPasswordFormComponent,
  RegisterFormComponent
} from '@app/authentication/components';

@NgModule({
  imports: [
    AuthRoutingModule,
    SharedModule,
    StoreModule.forFeature('auth', reducers),
    EffectsModule.forFeature([AuthEffects]),
    LangSwitcherModule,
    LogoModule,
    CivilityModule
  ],
  declarations: [
    AuthLayoutComponent,
    LoginViewComponent,
    SendResetPasswordViewComponent,
    ResetPasswordViewComponent,
    RegisterViewComponent,
    RegisterConfirmViewComponent,
    LoginFormComponent,
    ResetPasswordFormComponent,
    SendResetPasswordFormComponent,
    LogoutModalComponent,
    RegisterFormComponent
  ],
  entryComponents: [LogoutModalComponent],
  providers: [AuthFacade]
})
export class AuthModule {}
