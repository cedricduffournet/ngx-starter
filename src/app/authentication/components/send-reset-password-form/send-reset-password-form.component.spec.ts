import { TestBed, ComponentFixture } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';

import { TranslateModule } from '@ngx-translate/core';

import { FormModule } from '@app/shared/form';
import { SendResetPasswordFormComponent } from '@app/authentication/components';

describe('SendResetPasswordForm', () => {
  let fixture: ComponentFixture<SendResetPasswordFormComponent>;
  let component: SendResetPasswordFormComponent;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [ReactiveFormsModule, TranslateModule.forRoot(), FormModule],
      declarations: [SendResetPasswordFormComponent]
    });
    fixture = TestBed.createComponent(SendResetPasswordFormComponent);
    component = fixture.componentInstance;
  });

  it('should be created', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should not emit event if username is invalid', () => {
    const username = {
      username: 'test.com'
    };

    component.resettingForm.setValue(username);

    spyOn(component.resetPassword, 'emit');

    component.onReset();

    expect(component.resetPassword.emit).not.toHaveBeenCalled();
  });

  it('should not emit event if username is empty', () => {
    const username = {
      username: ''
    };

    component.resettingForm.setValue(username);

    spyOn(component.resetPassword, 'emit');

    component.onReset();

    expect(component.resetPassword.emit).not.toHaveBeenCalled();
  });

  it('should emit an event if the form is valid when submitted', () => {
    const username = {
      username: 'test@test.fr'
    };

    fixture.detectChanges();
    component.resettingForm.setValue(username);

    spyOn(component.resetPassword, 'emit');
    component.onReset();

    expect(component.resetPassword.emit).toHaveBeenCalledWith(username);
  });

  it('should disable submit button form if processing', () => {
    component.processing = true;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
