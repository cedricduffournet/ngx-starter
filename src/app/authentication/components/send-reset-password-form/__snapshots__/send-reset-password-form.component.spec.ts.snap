// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendResetPasswordForm should be created 1`] = `
<app-send-reset-password-form
  formBuilder={[Function FormBuilder]}
  processing="false"
  resetPassword={[Function EventEmitter]}
  resettingForm={[Function FormGroup]}
  submitted="false"
>
  <form
    class="ng-untouched ng-pristine ng-invalid"
    ng-reflect-form="[object Object]"
    novalidate=""
  >
    <app-text-input
      class="ng-untouched ng-pristine ng-invalid"
      formcontrolname="username"
      ng-reflect-help-text="HELP_FIELD_MAIL_RESET_PASSWORD"
      ng-reflect-label="EMAIL"
      ng-reflect-name="username"
      ng-reflect-placeholder="PLACEHOLDER_LOGIN"
      ng-reflect-submitted="false"
    >
      <div
        class="form-group "
      >
        <label
          class="control-label"
        >
          EMAIL
        </label>
        <input
          class="form-control"
          placeholder="PLACEHOLDER_LOGIN"
          type="text"
        />
        
        
        
        
        
        
        <small
          class="form-text text-muted"
        >
           HELP_FIELD_MAIL_RESET_PASSWORD 
        </small>
      </div>
    </app-text-input>
    <div
      class="form-group"
    >
      <button
        class="btn btn-primary btn-lg btn-block"
        ng-reflect-translate=""
        translate=""
        type="submit"
      >
         SEND_PASSWORD_RESET_MAIL 
      </button>
    </div>
  </form>
</app-send-reset-password-form>
`;

exports[`SendResetPasswordForm should disable submit button form if processing 1`] = `
<app-send-reset-password-form
  formBuilder={[Function FormBuilder]}
  processing={[Function Boolean]}
  resetPassword={[Function EventEmitter]}
  resettingForm={[Function FormGroup]}
  submitted="false"
>
  <form
    class="ng-untouched ng-pristine ng-invalid"
    ng-reflect-form="[object Object]"
    novalidate=""
  >
    <app-text-input
      class="ng-untouched ng-pristine ng-invalid"
      formcontrolname="username"
      ng-reflect-help-text="HELP_FIELD_MAIL_RESET_PASSWORD"
      ng-reflect-label="EMAIL"
      ng-reflect-name="username"
      ng-reflect-placeholder="PLACEHOLDER_LOGIN"
      ng-reflect-submitted="false"
    >
      <div
        class="form-group "
      >
        <label
          class="control-label"
        >
          EMAIL
        </label>
        <input
          class="form-control"
          placeholder="PLACEHOLDER_LOGIN"
          type="text"
        />
        
        
        
        
        
        
        <small
          class="form-text text-muted"
        >
           HELP_FIELD_MAIL_RESET_PASSWORD 
        </small>
      </div>
    </app-text-input>
    <div
      class="form-group"
    >
      <button
        class="btn btn-primary btn-lg btn-block"
        disabled=""
        ng-reflect-translate=""
        translate=""
        type="submit"
      >
         SEND_PASSWORD_RESET_MAIL 
      </button>
    </div>
  </form>
</app-send-reset-password-form>
`;
